File: 102-2_iOS程式設計/ImageIOSample02/ImageIOSample02/ViewController.m
//
//  ViewController.m
//  ImageIOSample02
//
//  Created by CSIEVNUT403 on 2014/5/22.
//  Copyright (c) 2014年 CSIEVNUT403. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)showImages
{
    NSURL *modelURL = [[NSBundle mainBundle] URLForResource:@"IMG_4079.JPG" withExtension:nil];
    
    CGImageSourceRef myImageSource = CGImageSourceCreateWithURL((__bridge CFURLRef) modelURL, NULL);
    
    CFDictionaryRef imageProperties = CGImageSourceCopyPropertiesAtIndex(myImageSource, 0, NULL);
    
    int imageSize = 240;
    CFNumberRef thumbnailSize = CFNumberCreate(NULL, kCFNumberIntType, &imageSize);
    CFTypeRef isCreateThumbnail = kCFBooleanTrue;
    
    NSString *createThumbnailaFromImage = (NSString *) kCGImageSourceCreateThumbnailFromImageIfAbsent;
    NSString *thumbnailMaxSize = (NSString *) kCGImageSourceThumbnailMaxPixelSize;
    
    NSDictionary *options = [[NSDictionary alloc] initWithObjectsAndKeys:(__bridge id)((void *) isCreateThumbnail), (void *) createThumbnailaFromImage, (void *) thumbnailSize, (void *) thumbnailMaxSize, nil];
    
    CGImageRef image = CGImageSourceCreateImageAtIndex(myImageSource, 0, NULL);
    CGImageRef thumbnailImage = CGImageSourceCreateImageAtIndex(myImageSource, 0, (__bridge CFDictionaryRef) options);
    
    [[self myImageView] setImage:[UIImage imageWithCGImage:image]];
    [[self myThumbnaiView] setImage:[UIImage imageWithCGImage:thumbnailImage]];
    
    CFRelease(myImageSource);
    CFRelease(imageProperties);
}

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
    [self showImages];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: 105-2_iOS程式設計/ImageView02/ImageView02/ViewController.swift
//
//  ViewController.swift
//  ImageView02
//
//  Created by mojado on 2017/4/5.
//  Copyright © 2017年 VNU. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    var arrayName = ["img01", "img02", "img03", "img04", "img05", "img06"]
    var arrayImage:Array<UIImage> = []
    var count:Int = 0

    @IBOutlet weak var imageShow: UIImageView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        imageShow.image = UIImage(named: "img01")
        count = arrayName.count
        
        for var i = 0; i < count; i++ {
            arrayImage.append(UIImage(named: arrayName[i])!)
        }
        
        imageShow.animationImages = arrayImage
        imageShow.animationDuration = NSTimeInterval(count * 2)
//        imageShow.animationRepeatCount = 0
    }

    @IBAction func startClick(sender: UIButton) {
        imageShow.startAnimating()
    }
    
    
    @IBAction func endClick(sender: UIButton) {
        imageShow.stopAnimating()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

